radixdlt/src/main/java/com/radixdlt/consensus/HashVerifier.java

```diff
@@ -15,24 +15,25 @@
  * language governing permissions and limitations under the License.
  */
 
-package org.radix.atoms.events;
+package com.radixdlt.consensus;
 
-import com.radixdlt.identifiers.EUID;
-
-import com.radixdlt.middleware2.CommittedAtom;
-import java.util.Set;
-import java.util.function.Supplier;
-
-public class AtomEventWithDestinations extends AtomEvent {
-	private Set<EUID> destinations;
-
-	public AtomEventWithDestinations(CommittedAtom atom, Supplier<Set<EUID>> destinationsSupplier) {
-		super(atom);
-		this.destinations = destinationsSupplier.get();
-	}
-
-	public Set<EUID> getDestinations() {
-		return destinations;
-	}
+import com.radixdlt.crypto.ECDSASignature;
+import com.radixdlt.crypto.ECPublicKey;
+import com.radixdlt.crypto.Hash;
 
+/**
+ * Verifies signatures against hashes.
+ */
+@FunctionalInterface
+public interface HashVerifier {
```

Comment:
Hmm you verify `signatures` not hashes, which also the JavaDoc implies...? `SignatureOfHashVerifier`? And also this should probably be moved to `radixdlt-java-common`? Should probably also be marked `SecurityCritical`? And be generalized as interface `Signing` is, and be implemented by `ECKeyPair`?

Reply:
Not part of PR